parsing/cLexer.ml4

```diff
@@ -513,16 +513,23 @@ let process_sequence loc bp c cs =
   aux 1 cs
 
 (* Must be a special token *)
-let process_chars loc bp c cs =
+let process_chars ?(diff_mode=false) loc bp c cs =
```

Comment:
Using an optional argument here and for the other function makes me worry that we're going to forget to pass it in some call. Why not make it required? Especially since it has no impact on external callers (the mli didn't change).

Reply:
I think `false` is the usual, "normal" setting and `true` will probably only ever be passed by the diff code.  Making it optional with a default implicitly documents this.  But I was not entirely consistent--`diff_mode` is a required argument for make_lexer but not the others.

I will change it if you still think I should.